backend/dao/usersDAO.js

```diff
@@ -0,0 +1,79 @@
+const ObjectID = require("mongoose").ObjectID;
+
+//This is a variable to use to store a reference to the db
+let users;
+
+//* injectDB method. (to initially conenct to db)
+//* as soon as server starts, get a reference to the users database.
+//* if it is already filled, then return, but if not already filled, then assign the variable with a reference to the db.
+module.exports = class UsersDAO {
+  static async injectDB(conn) {
+    if (users) {
+      return;
+    }
+    try {
+      //*connect to the db 'sample_analyrics' then access the collection 'customers'.
+      users = await conn.db(process.env.USER_NS).collection("customers");
+    } catch (e) {
+      console.error(`Unable to establish connection in usersDAO: ${e}`);
+    }
+  }
+  //Then, run the function get
+  //?call this to get a list of all users in the db
+  //?call this method to query/sort
+  //* corresponds to the .get
+  static async getUsers({
+    //can put many queries here..
+    //https://docs.mongodb.com/manual/reference/operator/
+    filters = null,
+    page = 0,
+    usersPerPage = 20,
+  } = {}) {
+    let query;
+    if (filters) {
+      if ("name" in filters) {
+        //this filter needed to be set up in Atlas, in 'create index'
+        query = { $text: { $search: filters["name"] } };
+      } else if ("email" in filters) {
+        query = { email: { $eq: filters["email"] } };
+      } else if ("accounts" in filters) {
+        query = { accounts: { $eq: filters["accounts"] } };
+      }
+    }
+    let cursor;
+    try {
+      cursor = await users.find(query); //*find all users from db that goes along with query that was passed in.
+    } catch (e) {
+      console.error(`Unable to issue find query: ${e}`);
+      return { usersList: [], totalNumUsers: 0 };
+    }
+    const displayCursor = cursor
+      .limit(usersPerPage) //* user per page
+      .skip((usersPerPage = page)); //*get to specific page of results

```

Comment:
Is there a reason why we are assigning `page` to `usersPerPage` here?
I'm against reassigning a new value to a parameter, I'm against any mutation that is not necessary too, mutations can lead to unexpected behavior, that leads to bugs.

Reply:
okay nice point, something I did not consider, thanks for offering another perspective for me

Reply:
Oh going back to look and run the code, the equal was a type, and it was meant to be a multiplier... but this was just to practice with the db, and I will most likely take the code out.. I think it is not necessary